test/realm_object_test.dart

```diff
@@ -314,4 +322,40 @@ Future<void> main([List<String>? args]) async {
   for (final pk in uuids) {
     testPrimaryKey(UuidPrimaryKey.schema, () => UuidPrimaryKey(pk), pk);
   }
+
+  test('Remapped property has correct names in Core', () {
+    final config = Configuration([RemappedClass.schema]);
+    final realm = getRealm(config);
+
+    final obj = realm.write(() {
+      final obj = realm.add(RemappedClass("some value"));
+      obj.listProperty.add(obj);
+      return obj;
+    });
+
+    final json = obj.toJson();
+
+    // remappedProperty is mapped as `__ other property __`
+    expect(json, contains('"__ other property __":"some value"'));
```

Comment:
I suggest we use better name for these. This code reads like a generated one.

Reply:
I wanted to try using some crazy names, but can make them more sensible.